### ISSUE TITLE
[BUG]: cannot unmarshal object into Go struct field Protection.lock_branch of type bool after upgrade to v5.13.0 

### ISSUE THREAD

**Issue by TheQueenIsDead** [integrations/terraform-provider-github#1476]
## Version

```console
$ terraform version
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/integrations/github v5.13.0
```

## Expected Behaviour

With Terraform config that could successfully execute a plan using module version v5.12.0.
I expect to be able to successfully execute a plan when upgrading to v5.13.0.

## Actual Behaviour

I receive an error unmarshalling an existing branch protection

```tf
│ Error: json: cannot unmarshal object into Go struct field Protection.lock_branch of type bool
│ 
│   with module.repository["<repo_name>"].github_branch_protection_v3.branch_protection[0],
│   on .terraform/modules/repository/main.tf line 240, in resource "github_branch_protection_v3" "branch_protection":
│  240: resource "github_branch_protection_v3" "branch_protection" {
│ 
```

## Diagnosis

As alluded to in this [comment](https://github.com/integrations/terraform-provider-github/pull/1415#issuecomment-1352424948), the module has recently updated go-github from 48.1.0 to 48.2.0 (https://github.com/integrations/terraform-provider-github/pull/1425).

I'm assuming that the SDK adding support for the `lock_branch` property from the API response (added [here](https://github.com/google/go-github/pull/2577/files#diff-7a0544e44d81e972917728b13efd3fa55e7f9d1bd8edb745216494fa1e178f93)), is causing the unmarshalling to fail, and this could perhaps be fixed by adding a `lock_branch` property to the branch protection state

**Comment by hero-david** [comment]
As @kfcampbell mentioned, this is resolved due to:

> https://github.com/google/go-github/issues/2612, fixed in https://github.com/google/go-github/pull/2606, released [here](https://github.com/google/go-github/releases/tag/v49.0.0), and added to the project as part of https://github.com/integrations/terraform-provider-github/pull/1473.

**Comment by TheQueenIsDead** [comment]
As @kfcampbell mentioned, this is resolved for me in the 5.14.0 release due to:

> https://github.com/google/go-github/issues/2612, fixed in https://github.com/google/go-github/pull/2606, released [here](https://github.com/google/go-github/releases/tag/v49.0.0), and added to the project as part of https://github.com/integrations/terraform-provider-github/pull/1473.
